hfile.c

```diff
@@ -212,6 +214,7 @@ int hfile_set_blksize(hFILE *fp, size_t bufsiz) {
 }
 
 /* Called only from hgetc(), when our buffer is empty.  */
+HTSLIB_EXPORT
```

Comment:
Why does this method need to be exported, since it is only used internally by `hread`, inside which it is already declared `extern`?
Same for `hread2`, `hputc2`, `hwrite2`, `hputs2`.

Reply:
It's declared in this way because it needs to be called from external code but only via `hread` and not directly.  Declaring it `extern` isn't enough to have it exported - it needs the annotation, which has to be in the C file as the only declaration in the header file is hidden in the `hread` function definition.

Reply:
What's missing from the above explanation is that it's called from a static inline, and thus external code is essentially calling these internal functions (hence the HTSLIB_EXPORT).

However that means hread2 etc are also part of the external API and external ABI and should be documented as such.  The comment "Called only from hgetc()" is misleading - it's only called by our code there, but people are free to call it anywhere as it is a public function.

Reply:
`hgetc2`/`hread2`/etc are part of the **ABI** (as you say, as they are invoked by an **inline** API function so calls to these functions are directly burnt into client executables), but they are not part of the **API** (as there is no file-scope declaration of them in a public header). The comment is accurate as the only declaration of `hgetc2()` has a scope limited to the body of the `hgetc()` function — so it truly can only be called from that function.

People could call it from anywhere by writing their own declaration for it, but that would be cheating, could be done for any (visible) function, and is not an activity that HTSlib (or any sensible library) supports.

Reply:
Fair enough.   Makes sense now I look again.

Thanks.